alpine/entry.sh

```diff
@@ -1,11 +1,17 @@
 #!/bin/bash
 
+# On ResinOS 2.x devices, the hostname is set by the hostOS.
+# For backward compatibility, we only update the hostname for ResinOS 1.x devices.
 function update_hostname()
 {
-	HOSTNAME="$HOSTNAME-${RESIN_DEVICE_UUID:0:7}"
-	echo $HOSTNAME > /etc/hostname
-	echo "127.0.1.1 $HOSTNAME" >> /etc/hosts
-	hostname "$HOSTNAME"
+	HOSTNAME=$(hostname)
+	if [ ${RESIN_DEVICE_UUID:0:7} != ${HOSTNAME:0:7} ]; then
+		# For 1.x Devices only.
+		HOSTNAME="$HOSTNAME-${RESIN_DEVICE_UUID:0:7}"
```

Comment:
Can't we base the hostname only to variables that are set by the supervisor? For example `HOSTNAME="$RESIN_DEVICE_TYPE-${RESIN_DEVICE_UUID:0:7}"`